pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,56 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+	regionCfg := &aws.Config{Region: aws.String(dsInfo.Region)}
+	cfgs := []*aws.Config{
+		regionCfg,
+	}
+	// Choose authentication scheme based on the type chosen for the data source
+	// Basically, we support the following methods:
+	// Shared credentials: Providing access key pair sourced from user's AWS credentials file
+	// Static credentials: Providing access key pair directly
+	// SDK: Leave it to SDK to decide
+	switch dsInfo.AuthType {
+	case "credentials":
+		plog.Debug("Authenticating towards AWS with shared credentials", "profile", dsInfo.Profile,
```

Comment:
These debug log statement could have a performance impact since our logging is slow and/or create a lot of spam in logs - we have users running Grafana with all debug messages enabled. Would suggest removing these since they will be printed for every query if I'm not mistaken.

Reply:
@marefr Now that a session cache is introduced, do you still think logging here is a problem?

Reply:
Guess not